javagen/src/main/java/com/azure/autorest/mapper/ModelPropertyMapper.java

```diff
@@ -170,6 +170,14 @@ public ClientModelProperty map(Property property) {
         }
 
         // handle x-ms-client-default for primitive type, enum, boxed type and string
+        if (property.getClientDefaultValue() == null) {
+            if (property.getSchema() != null && property.getSchema() instanceof ConstantSchema) {
+                ConstantSchema constantSchema = (ConstantSchema) property.getSchema();
+                if (constantSchema.getValue() != null) {
+                    property.setClientDefaultValue(constantSchema.getValue().getValue().toString());
```

Comment:
Don't set it back, just use your result. `Property` is data model, consider it as immutable.